### PR TITLE
feat: initial release

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
### Description

The initial push of the key protect key ring module

**Check the relevant boxes:**
- [ ] Bug fix (nonbreaking change that fixes an issue)
- [x] New feature (nonbreaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [ ] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,159 +1,82 @@
 <!-- BEGIN MODULE HOOK -->
 
 <!-- Update the title to match the module name and add a description -->
-# Terraform Modules Template Project
+# Terraform IBM -key-protect-key-ring
 <!-- UPDATE BADGE: Update the link for the following badge-->
-[![Incubating (Not yet consumable)](https://img.shields.io/badge/status-Incubating%20(Not%20yet%20consumable)-red)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
-[![Build status](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/actions/workflows/ci.yml/badge.svg)](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/actions/workflows/ci.yml)
+[![Stable (With Quality Checks)](https://img.shields.io/badge/Status-Stable%20(With-Quality-Checks)-yellowgreen?style=plastic)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
```
Comment (daniel-butler-irl) [inline README.md:6]: This badge is broken...

File: README.md [hunk 2]
```diff
@@ -1,159 +1,82 @@
 <!-- BEGIN MODULE HOOK -->
 
 <!-- Update the title to match the module name and add a description -->
-# Terraform Modules Template Project
+# Terraform IBM -key-protect-key-ring
```
Comment (ocofaigh) [inline README.md:4]: Terraform IBM Key Protect Key Ring module

File: variables.tf [hunk 3]
```diff
@@ -2,10 +2,23 @@
...
 
-#variable "myvariable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = ""
-#}
+variable "endpoint_type" {
+  type        = string
+  description = "The type of endpoint to be used for creating keys, accepts 'public' or 'private'"
+  default     = "public"
```
Comment (ocofaigh) [inline variables.tf:8]: should we not default this to private?
Also lets add validation on the inputs here?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see minor comments

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**